repos/spack_repo/builtin/packages/vtk/package.py

```diff
@@ -90,6 +86,140 @@ class Vtk(CMakePackage):
+        depends_on("pegtl@3", when="@:9.5.2")
+        depends_on("scnlib")
+        depends_on("verdict")
+
+        # currently excluded
+        # depends_on("viskores")
+        # depends_on("token")
+        # depends_on("exprtk")
+
+    # TODO IOLAS, InfoVis/Boost, InfoVis/BoostGraph and FilterReebGraph
+    depends_on(Boost.with_default_variants)
+
+    # === 2nd-level DEPENDENCIES, for 3rd party libraries built internally
+    with when("@:8 io=xdmf"):
+        depends_on("mpi", when="+mpi")
+
+        depends_on("boost")
+        depends_on("boost+mpi", when="+mpi")
+
+        # TODO: replace this with an explicit list of components of Boost,
+        # for instance depends_on('boost +filesystem')
+        # See https://github.com/spack/spack/pull/22303 for reference
+        depends_on(Boost.with_default_variants, when="io=xdmf")
+
+    depends_on("mpi", when="+mpi")  # for builing viskores internally
+
+    # TODO just for theora, check when theora stopped being built internally
+    depends_on("libogg", when="@:8")
+
+    # === CONDITIONAL DEPENDENCIES
+    with when("io=adios2"):
+        depends_on("adios2+mpi", when="+mpi")
+        depends_on("adios2~mpi", when="~mpi")
+    requires("io=adios2", when="@:8 io=fides")
+
+    depends_on("ffmpeg", when="io=ffmpeg")
+
+    with when("io=cgns"):
+        with when("@9:"):
+            depends_on("cgns@4.1.1:")
+            depends_on("cgns+mpi", when="+mpi")
+            depends_on("cgns~mpi", when="~mpi")
+
+    # VTK introduced Seacas IOSS dependency on 9.1
+    with when("@9.1: io=ioss"):
+        depends_on("seacas+mpi", when="+mpi")
+        depends_on("seacas~mpi", when="~mpi")
+
+        depends_on("seacas@2021-05-12:2022-10-14", when="@9.1")
+        # vtk@9.2: need Ioss::Utils::get_debug_stream() which only 2022-10-14 provides,
+        # and to be safe against other issues, make them build with this version only:
+        depends_on("seacas@2022-10-14", when="@9.2:9.3")
+        depends_on("seacas@2024-06-27", when="@9.4:")
+        # TODO probably we can bump version
+
+    with when("@9: io=netcdf"):
+        depends_on("netcdf-c ~mpi", when="~mpi")
+        depends_on("netcdf-c +mpi", when="+mpi")
+        depends_on("netcdf-c@:4.9.2")
+    depends_on("netcdf-cxx4", when="io=netcdf @:8.1.2")
+
+    depends_on("ospray@2.1:2", when="raytracing=ospray")
+    depends_on("openimagedenoise", when="raytracing=ospray")
+    depends_on("ospray +mpi", when="raytracing=ospray +mpi")
+
+    # VTK will need Qt5OpenGL, and qt needs '-opengl' for that
+    depends_on("qt+opengl", when="+qt")
+
+    # === TODO TO BE CHECKED
+
+    # The use of the OpenGL2 backend requires at least OpenGL Core Profile
+    # version 3.2 or higher.
+    depends_on("gl@3.2:", when="+opengl2")
+    depends_on("gl@1.2:", when="~opengl2")
+
+    depends_on("xz")
+    depends_on("libxt", when="^[virtuals=gl] glx platform=linux")  # just for x11? web?
+    depends_on("glew")  # viskores
```

Comment:
is this just for `viskores` (ex `vtkm`)? just double checking so that when I will make `viskores` external I can drop it in newer release.